sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,261 @@
+                             "must be at least 2."
+                             .format(indices))
+        return n_bins
+
+    @staticmethod
+    def _check_ignored_features(ignored, n_features):
+        if ignored is None:
+            return np.array([], dtype='int64')
+
+        ignored = check_array(ignored, ensure_2d=False, dtype=int)
+        ignored = column_or_1d(ignored)
+
+        if len(set(ignored)) != ignored.shape[0]:
+            raise ValueError("Duplicate ignored column indices found.")
+
+        if np.all(ignored >= 0) and np.all(ignored < n_features):
+            return ignored
+
+        raise ValueError("Invalid ignored feature index.")
+
+    def transform(self, X):
+        """Discretizes the data.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data to be discretized.
+
+        Returns
+        -------
+        Xt : numeric array-like, shape (n_samples, n_features)
+            Data in the binned space.
+        """
+        X = self._check_X_post_fit(X)
+
+        return _transform_selected(X, self._transform,
+                                   self.transformed_features_, copy=True,
+                                   retain_order=True)
+
+    def _transform(self, X):
+        """Performs transformation on X, with no ignored features."""
+        trans = self.transformed_features_
+
+        X -= self.offset_[trans]
+        bin_width = self.bin_width_[trans]
+
+        # Rescale into [-1, bin_width] range
+        with np.errstate(divide='ignore', invalid='ignore'):
+            X /= bin_width
+
+        # Values which are a multiple of the bin width are susceptible to
+        # numeric instability. Add eps to X so these values are binned
+        # correctly. See documentation for numpy.isclose for an explanation
+        # of ``rtol`` and ``atol``.
+        rtol = 1.e-5
+        atol = 1.e-8
+        eps = atol + rtol * abs(bin_width)
+        np.floor(X + eps, out=X)
+
+        X[~np.isfinite(X)] = 0  # Case when a feature is constant
+        np.clip(X, 0, self.n_bins_[trans] - 1, out=X)
+        return X
+
+    def inverse_transform(self, Xt):
+        """Transforms discretized data back to original feature space.
+
+        Parameters
+        ----------
+        Xt : numeric array-like, shape (n_sample, n_features)
+            Transformed data in the binned space.
+
+        Returns
+        -------
+        Xinv : numeric array-like
+            Data in the original feature space.
+        """
+        Xt = self._check_X_post_fit(Xt)
+        trans = self.transformed_features_
+        Xinv = Xt.copy()
+        Xinv_sel = Xinv[:, trans]
+
+        Xinv_sel += 0.5
+        Xinv_sel *= self.bin_width_[trans]
+        Xinv_sel += self.offset_[trans]
+
+        Xinv[:, trans] = Xinv_sel
+        return Xinv
+
+    def _check_X_post_fit(self, X):
+        check_is_fitted(self, ["offset_", "bin_width_"])
```

Comment:
Hmm, it's a little problematic that we are indeed checking `check_is_fitted(self, ["offset_", "bin_width_"])`, which is inconsistent with the function's name.

The reason why I put this into its own function is because these checks were being used in two places. But maybe `_check_post_fit` suffices.